src/data_structures/src/array_ext.cairo

```diff
@@ -385,4 +391,26 @@ impl SpanImpl<T, impl TCopy: Copy<T>, impl TDrop: Drop<T>> of SpanTraitExt<T> {
 
         ret
     }
+
+    fn unique<impl TPartialEq: PartialEq<T>>(mut self: Span<T>) -> Array<T> {
+        let mut ret = array![];
+        loop {
+            let shortest: Span<T> = if (self.len() <= 1) || (ret.len() <= self.len()) {
+                ret.span()
+            } else {
+                self
+            };
+            match self.pop_front() {
+                Option::Some(v) => {
+                    if !shortest.contains(*v) {
+                        ret.append(*v);
+                    }
+                },
+                Option::None => {
+                    break;
+                }
+            };
+        };
+        ret
+    }
```

Comment:
This pass your tests on my end:
```suggestion
        let mut ret = array![];
        loop {
            match self.pop_front() {
                Option::Some(v) => {
                    if !ret.contains(*v) {
                        ret.append(*v);
                    }
                },
                Option::None => {
                    break;
                }
            };
        };
        ret
    }
```

Reply:
Yea indeed, I had this at first but I was trying to follow the recommendations in the issue:
> Loop over the array, pop_front elements one by one and use contains to identifiy if the element already exists in the shortest array (input or ouput) then add the element in the ouput array.

Thought it would have some wizard optimization magic that I missed.. 😅